docs/developer/how-to/make-new-ophyd-async-device.md

```diff
@@ -0,0 +1,58 @@
+# Decision Flowchart for Creating a New ophyd_async Device
+
+This document contains decision flowcharts designed to guide developers through the process of creating a new ophyd_async device in the Ophyd library. These flowcharts help in determining the appropriate class inheritance, methods to override, and testing procedures for optimal device functionality and integration into the system.
+
+## High-Level Development Flowchart
+
+This high-level flowchart guides the overall process of creating a new ophyd_async device, from testing PVs to overriding methods and determining state enums.
+
+```{mermaid}
+
+
+  flowchart TD
+    highLevelStart([Start]) --> scoutUsers[Scout current and potential users of the device, tag them in the draft PR]
+    scoutUsers --> testPVs[Test PVs to get the right values]
+    testPVs --> decideStateEnums[Decide on State Enums: extend str, Enum]
+    decideStateEnums --> chooseMethods[Choose which superclass methods to override]
+    chooseMethods --> finalizeDevice[Finalize Device Implementation]
+    finalizeDevice --> requestFeedback[Request feedback from tagged users]
+    requestFeedback --> rebaseOnMain[Rebase on main -> make sure tests pass still]
+    rebaseOnMain --> coordinateMerges[Coordinate merges with other codebase updates]
+    coordinateMerges --> markPRReady[Mark PR as ready]
```

Comment:
Could: I'm actually not sure this works very well as  a flowchart for the following reasons:
1. It prescribes an arbitrary order in which to do things, who says I should test the PVs before coming up with enum values? The other way around is actually easier using the dodal CLI. 
2. It implies you should do each step once and never again. In reality making a device is nothing like that, it is an iterative process. The dodal CLI makes it easy to  constantly test the PVs as you write signals, rather than once upfront. You also don't come up with a list of enums upfront, you work through one signal at a time and when you discover one that needs to be an enum, you write it and decide on enum values for that signal only, rinse and repeat.
3. It takes up a lot of page-space to convey comparatively little information.

I think what we want here is a worked example, rather than a precise series of steps. Perhaps a paragraph explaining in prose about testing the PVs, making enum signals, raising draft PRs etc. and then an example device with all of those things?

Reply:
I agree with Callum, I think a worked example would be better

Reply:
> It takes up a lot of page-space to convey comparatively little information.
1 I got this based on #480  experiences where defining enum values before testing the right PVs was counterproductive. I'm happy to change this into a command from dodal cli, I didn't know this existed
2 then we can change it for there to be an iterative process.  we can put what you just wrote.
page-space is free and it's not like we're printing this. empty space is the key to clear design. 

re 'worked example': it's personal preference, it's fair that some people prefer prose modality to a diagram or the other way around, I think for inclusiveness we should support both.

Reply:
I would argue they are fundamentally different. My suggestion is a rough guide to help you figure out the sequence of steps you need to take. The current flowchart provides a precise sequence of steps, which I think is too prescriptive.

Reply:
Of the 3 flowcharts, only one has branching while the others are a strict linear progression. I agree that those make more sense as a worked example.

Reply:
@callumforrester pls what would be a good prose exmple here?

Reply:
This but with more words/detail:

"Familiarise yourself with the basics of ophyd-async devices <link>, some considerations you should make while developing are <blank>, see example below. You are advised to gather requirements early, raise draft PRs, seek feedback, make use of the CLI to test. You should try to consolidate similar devices, the tests give you confidence that they will still work if you change them
<EXAMPLE>".

Reply:
> The current flowchart provides a precise sequence of steps, which I think is too prescriptive.

this is just a guide and of course every user will need to adapt this. word-by-word maybe less than 5% of devices would have gone through this process *in this step order*. 

I don't see this as 'the order' to do things, just 'an order of things to base your investigations on that roughly should work if your use case is the average one'.

and I'm not sure how many more words/detail you see fit, this is very vague to me maybe I am not the best person to write out the prose part.